modules/network/src/test/java/org/apache/ignite/internal/network/StaticNodeFinderTest.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.network;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.InetAddress;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import org.apache.ignite.internal.testframework.WorkDirectory;
+import org.apache.ignite.internal.testframework.WorkDirectoryExtension;
+import org.apache.ignite.network.NetworkAddress;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(WorkDirectoryExtension.class)
+class StaticNodeFinderTest {
+    @WorkDirectory
+    private Path workDir;
+
+    @Test
+    void returnsIpAddresses() {
+        NetworkAddress ipv4 = new NetworkAddress("1.2.3.4", 3001);
+        NetworkAddress ipv6 = new NetworkAddress("2001:1234:130f:1234:1234:99c0:876a:130b", 3002);
+        NodeFinder finder = new StaticNodeFinder(List.of(ipv4, ipv6));
+
+        assertThat(finder.findNodes(), contains(ipv4, ipv6));
+    }
+
+    @Test
+    void resolvesLocalHostToJustOneAddress() throws Exception {
+        NodeFinder finder = new StaticNodeFinder(List.of(new NetworkAddress("localhost", 3001)));
```

Comment:
How do you guarantee that there's more than one address to choose from?

Reply:
I rewrote it to the approach with overrides to guarantee this